corehq/apps/app_manager/views/modules.py

```diff
@@ -716,33 +719,38 @@ def delete_module(request, domain, app_id, module_unique_id):
         m.unique_id for m in app.get_modules()
         if m.module_type == 'shadow' and m.source_module_id == module_unique_id and m.root_module_id is not None
     ]
-    deletion_records = []
     for shadow_child in shadow_children:
         # We don't allow directly deleting or editing the source of a shadow
         # child, so we delete it when its source is deleted
         deletion_records.append(app.delete_module(shadow_child))
 
     deletion_records.append(app.delete_module(module_unique_id))
-    for record in deletion_records:
-        messages.success(
-            request,
-            _('You have deleted "{name}". <a href="{url}" class="post-link">Undo</a>').format(
-                name=record.module.default_name(app=app),
-                url=reverse('undo_delete_module', args=[domain, record.get_id])
-            ),
-            extra_tags='html'
-        )
+    restore_url = "{}?{}".format(
+        reverse('undo_delete_module', args=[domain]),
+        "&".join("record_id={}".format(record.get_id) for record in deletion_records)
+    )
+    deleted_names = ", ".join(record.module.default_name(app=app) for record in deletion_records)
+    messages.success(
+        request,
+        _(f'You have deleted "{deleted_names}". <a href="{restore_url}" class="post-link">Undo</a>'),
```

Comment:
translate, please

Reply:
It is already translated (`_(f'...`)

Reply:
I don't think that'll work: the string passed to `_` will contain the substitutions, and that string won't have a translation.

(I would have written a more helpful comment in the first place but I hadn't noticed the `_`.)